pkg/types/attributes.go

```diff
@@ -338,32 +340,38 @@ func (a Attributes) NilSafeSet(key string, value interface{}) {
 // GetAsString gets the string value for the given key (+ `true`),
 // or empty string (+ `false`) if none was found
 func (a Attributes) GetAsString(key string) (string, bool) {
-	// check in predefined attributes
-	if value, found := Predefined[key]; found {
-		return value, true
-	}
 	if value, found := a[key]; found {
+		if value == nil {
+			return "", false // nil here means attribute was reset
+		}
 		if value, ok := value.(string); ok {
 			return value, true
 		} else if v, ok := a[key]; ok {
 			return fmt.Sprintf("%v", v), true
 		}
 	}
+	// check in predefined attributes
```

Comment:
I guess that storing a `nil` value instead of removing it is because you also changed the priority for looking up attributes? We don't want to look in the predefined attributes if there was an explicit reset beforehand?

Reply:
Yes.  I need a way to distinguish something that was *reset* vs something that was simply not assigned.

Reply:
yes, ok, it makes sense.